### PR TITLE
feat: implement ES2025 Uint8Array base64/hex methods

### PR DESCRIPTION
**Pull request by TooTallNate**
## Summary

- Implements the full ES2025 `Uint8Array` base64/hex API (proposal-arraybuffer-base64, Stage 4)
- All 6 methods implemented entirely in C (`source/uint8array.c`) using mbedtls for base64 encoding and a custom spec-compliant decoder for all decode paths
- TypeScript type declarations provided since TypeScript doesn't ship these types yet (not even in 5.9.3)

## Methods

| Method | Description |
|---|---|
| `Uint8Array.prototype.toBase64(options?)` | Encode to base64 string. Options: `alphabet` (`"base64"` / `"base64url"`), `omitPadding` |
| `Uint8Array.prototype.toHex()` | Encode to lowercase hex string |
| `Uint8Array.prototype.setFromBase64(string, options?)` | Decode base64 into existing array. Returns `{read, written}`. Options: `alphabet`, `lastChunkHandling` |
| `Uint8Array.prototype.setFromHex(string)` | Decode hex into existing array. Returns `{read, written}` |
| `Uint8Array.fromBase64(string, options?)` | Decode base64 to new Uint8Array. Options: `alphabet`, `lastChunkHandling` |
| `Uint8Array.fromHex(string)` | Decode hex to new Uint8Array |

## Spec compliance

- ASCII whitespace stripping in base64 input (TAB, LF, FF, CR, SPACE)
- `lastChunkHandling`: `"loose"` (default), `"strict"`, `"stop-before-partial"`
- Proper `read` tracking (only updated after full 4-char chunks per spec)
- Early return when `maxLength` is 0
- `TypeError` on non-string input, `SyntaxError` on invalid encoding
- `setFromBase64`/`setFromHex` partial-fill semantics with `{read, written}` result

## Tests

55 tests in `apps/tests/src/uint8array.test.ts` covering all MDN examples, all options, round-trips, and edge cases from the ECMAScript spec.

### CODE REVIEW

File: source/uint8array.c [hunk 1]
```diff
@@ -0,0 +1,651 @@
...
+			}
+			written += 3;
+			chunk_len = 0;
+			read = i;  // update read after successful full chunk
+			if (written == max_output) {
+				*out_read = read;
+				*out_written = written;
+				return 0;
+			}
+		}
```
[the remark this answers is not in the document]
Comment (TooTallNate) [reply source/uint8array.c]: Fixed. Added `if (written > SIZE_MAX - 3)` guard before the full chunk decode in `base64_decode_impl()`. In practice this is unreachable since `max_output` bounds the loop, but it's good defense-in-depth.

File: packages/runtime/src/crypto.ts [hunk 2]
```diff
@@ -669,7 +669,16 @@ function getHashLength(hash: HashAlgorithmIdentifier): number {
 
 type HashAlgorithmIdentifier = AlgorithmIdentifier;
 
-const { base64urlEncode, base64urlDecode } = $;
+const b64urlOpts = { alphabet: 'base64url' as const };
+
+function base64urlEncode(buf: ArrayBuffer): string {
+	return new Uint8Array(buf).toBase64(b64urlOpts);
+}
```
[the remark this answers is not in the document]
Comment (TooTallNate) [reply packages/runtime/src/crypto.ts]: Good catch — real bug. The old native `base64urlEncode` stripped padding, but the replacement `toBase64({ alphabet: 'base64url' })` did not. Added `omitPadding: true` to `b64urlOpts`. JWK base64url (RFC 7515/7517) requires unpadded encoding.

File: source/uint8array.c [hunk 3]
```diff
@@ -0,0 +1,675 @@
...
+				*out_read = read;
+				*out_written = written;
+				JS_ThrowSyntaxError(ctx, "Invalid base64: non-zero overflow bits");
+				return -1;
+			}
+			if (output) {
+				output[written]     = (chunk[0] << 2) | (chunk[1] >> 4);
+				output[written + 1] = ((chunk[1] & 0xF) << 4) | (chunk[2] >> 2);
+			}
+			written += 2;
```
[the remark this answers is not in the document]
Comment (TooTallNate) [reply source/uint8array.c]: Fixed. Added `if (written > SIZE_MAX - 1)` and `if (written > SIZE_MAX - 2)` guards before the padded final-chunk increments.
